avocado/plugins/multiplex.py

```diff
@@ -35,12 +35,10 @@ def __init__(self, *args, **kwargs):
         super(Multiplex, self).__init__(*args, **kwargs)
 
     def configure(self, parser):
-        if multiplexer.MULTIPLEX_CAPABLE is False:
-            return
-
         parser = super(Multiplex, self).configure(parser)
-        parser.add_argument('multiplex', nargs='+',
-                            help='Path(s) to a multiplex file(s)')
+        if multiplexer.MULTIPLEX_CAPABLE:
+            parser.add_argument("-m", '--multiplex', nargs='*',
```

Comment:
Let me check if I got this right: "it should allow coexisting of different plugins at the same time".

You mean that multiple providers could feed the same Multiplexer instance?  This is certainly the most flexible and powerful approach, but I have to say I'm a bit skeptical towards committing to it. Even though that `data_merge()` seems to be a big part of the necessary support, I bet there are other hidden issues to support that.

If you feel secure that we can achieve this level of flexibility, then please ignore my skepticism.  But, I'd ask you to consider having a secondary feeder implementation to the Mux, as a way to prove the design and implementation is sound.

With regards to the arguments, if the multiple "feeder" design is the way to go, I'd go with keeping the "old" `--multiplex` an alias to `--mux-yaml`, but with a deprecation notice.  IMHO we shouldn't overwrite a parameter such as `--multiplex` as be silent about it.

The other approach wrt arguments, which I dislike, would be to have generic arguments, such as the test URLs, and give all plugins a change to use those.  But that would be unnecessarily complex IMHO.